apis/Google.Cloud.Firestore.Data/Google.Cloud.Firestore.Data/FirestoreDb.cs

```diff
@@ -148,5 +153,44 @@ internal DocumentReference GetDocumentReferenceFromResourceName(string reference
                 nameof(referenceValue), "Reference {0} is not part of database {1}", referenceValue, RootPath);
             return Document(referenceValue.Substring(DocumentsPath.Length + 1));
         }
+
+        /// <summary>
+        /// Fetches document snapshots from the server, based on an optional transaction ID.
+        /// </summary>
+        /// <param name="documents">The document references to fetch. Must not be null.</param>
+        /// <param name="transactionId">A transaction ID, or null to not include any transaction ID.</param>
+        /// <param name="cancellationToken">A cancellation token for the operation.</param>
+        /// <returns>The document snapshots, in the order they are provided in the response. (This may not be the order of <paramref name="documents"/>.)</returns>
+        internal async Task<IList<DocumentSnapshot>> GetDocumentSnapshotsAsync(IEnumerable<DocumentReference> documents, ByteString transactionId, CancellationToken cancellationToken)
+        {
+            var request = new BatchGetDocumentsRequest { Database = RootPath, Documents = { documents.Select(d => d.Path) } };
+            if (transactionId != null)
+            {
+                request.Transaction = transactionId;
+            }
+            var stream = Client.BatchGetDocuments(request, CallSettings.FromCancellationToken(cancellationToken));
+            var responseStream = stream.ResponseStream;
```

Comment:
Should `responseStream` be disposed of?

Reply:
Good question. Um, probably? Will do.

Reply:
The docs deep in gRPC have:

>  Normally, there is no need for you to dispose the call unless you want to utilize the "Cancel" semantics of invoking `Dispose`.

But that's probably worth doing anyway.